docs/docs/alephium/entrypoints.md

```diff
@@ -0,0 +1,652 @@
+
+#### Input parameters
+
+| Name     | Type      | Description                                              |
+| -------- | --------- | -------------------------------------------------------- |
+| owner_id | AccountId | An AccountId identifying the user who own the positions. |
+
+#### Output parameters
+
+| Type           | Description                             |
+| -------------- | --------------------------------------- |
+| Vec<Position\> | A list containing the user's positions. | -->
+
+## Swap
+
+### Swap
+
+```rust
+@using(preapprovedAssets = true, checkExternalCaller = false)
+pub fn swap(
+    poolKey: PoolKey,
+    xToY: Bool,
+    amount: U256,
+    byAmountIn: Bool,
+    sqrtPriceLimit: U256
+) -> CalculateSwapResult
+```
+
+This function executes a swap based on the provided parameters. It transfers tokens from the user's address to the contract's address and vice versa. The swap will fail if the user does not have enough tokens, has not approved enough tokens, or if there is insufficient liquidity.
+
+#### Input parameters
+
+| Name             | Type        | Description                                                                             |
+| ---------------- | ----------- | --------------------------------------------------------------------------------------- |
+| poolKey         | PoolKey     | Pool key of the pool on which you wish to perform the swap.                             |
+| xToY           | Bool        | Specifies the direction of the swap.                                                    |
+| amount           | U256 | Amount of tokens you want to receive or give.                                           |
+| byAmountIn     | Bool        | Indicates whether the entered amount represents the tokens you wish to receive or give. |
+| sqrtPriceLimit | U256   | If the swap achieves this square root of the price, it will be canceled.                |
+
+#### Output parameters
+
+| Type                | Description                                                                                                                                  |
+| ------------------- | -------------------------------------------------------------------------------------------------------------------------------------------- |
+| CalculateSwapResult | A struct containing the amount in and amount out with starting and target square root of price, taken fee, pool. |
+
+#### Errors
+
+| Code                | Description                                                                                                         |
+| ------------------- | ------------------------------------------------------------------------------------------------------------------- |
+| `ZeroAmount`      | Fails if the user attempts to perform a swap with zero amounts.                                                     |
+| `WrongPriceLimit` | Fails if the square root of price or price limit is incorrect. |
```

Comment:
you can add that the limit direction was incorrectly specified and regardless of the swap outcome, it will be invalid due to this limit